packages/cli/src/cmds/beacon/initBeaconState.ts

```diff
@@ -112,26 +112,23 @@ export async function initBeaconState(
       throw Error(`Must set arg --weakSubjectivityServerUrl for network ${args.network}`);
     }
 
-    let stateId = "finalized";
-    let checkpoint: Checkpoint | undefined;
+    let checkpoint: Checkpoint;
     if (args.weakSubjectivityCheckpoint) {
       checkpoint = getCheckpointFromArg(args.weakSubjectivityCheckpoint);
-      stateId = (checkpoint.epoch * SLOTS_PER_EPOCH).toString();
+    } else {
+      const checkpointUrl = `${remoteBeaconUrl}/eth/v1/beacon/states/finalized/finality_checkpoints`;
+      logger.info("Fetching finalized checkpoint", {checkpointUrl});
+      checkpoint = await fetchFinalizedCheckpoint(checkpointUrl);
     }
-    const url = `${remoteBeaconUrl}/eth/v1/debug/beacon/states/${stateId}`;
-
+    const stateSlot = checkpoint.epoch * SLOTS_PER_EPOCH;
+    const apiVersion = chainForkConfig.getForkName(stateSlot) === ForkName.phase0 ? "v1" : "v2";
+    const url = `${remoteBeaconUrl}/eth/${apiVersion}/debug/beacon/states/${stateSlot}`;
     logger.info("Fetching weak subjectivity state from " + url);
+
     const wsState = await fetchWeakSubjectivityState(chainForkConfig, url);
     const config = createIBeaconConfig(chainForkConfig, wsState.genesisValidatorsRoot);
     const store = lastDbState ?? wsState;
-    return initAndVerifyWeakSubjectivityState(
-      config,
-      db,
-      logger,
-      store,
-      wsState,
-      checkpoint || getCheckpointFromState(config, wsState)
-    );
+    return initAndVerifyWeakSubjectivityState(config, db, logger, store, wsState, checkpoint);
```

Comment:
All this logic would be better placed if in `fetchWeakSubjectivityState` I think, including the fetch of the checkpoint. By printing the checkpoint epoch + the lodestar version it's enough info to know that URL was used